src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -154,171 +153,176 @@ private RootCommand SetupCommandLineAPI()
-                // Find files to analyze
-                var filesToAnalyze = new List<FileInfo>();
-                FindJsonFilesInDirectoryRecursive(directoryPath, filesToAnalyze);
+                ConfigurationDefinition config = ReadConfigurationFile(configurationFilePath);
+                if (config == null)
+                    return ExitCode.ErrorGeneric;
 
-                // Log root directory info to be analyzed
-                Console.WriteLine(Environment.NewLine + Environment.NewLine + $"Directory: {directoryPath}");
-
-                int numOfFilesAnalyzed = 0;
-                bool issueReported = false;
-                var filesFailed = new List<FileInfo>();
-                foreach (FileInfo file in filesToAnalyze)
-                {
-                    int res = AnalyzeTemplate(file, null, configurationsFilePath, reportFormat, outputFilePath, runTtk, verbose, false, reportWriter, false, logger);
-                    if (res == (int)ExitCode.Success)
-                    {
-                        numOfFilesAnalyzed++;
-                    }
-                    else if (res == (int)ExitCode.Violation)
-                    {
-                        numOfFilesAnalyzed++;
-                        issueReported = true;
-                    }
-                    else if (res == (int)ExitCode.ErrorGeneric)
-                    {
-                        filesFailed.Add(file);
-                    }
-                }
-
-                Console.WriteLine(Environment.NewLine + $"Analyzed {numOfFilesAnalyzed} file(s).");
-                if (filesFailed.Count > 0)
-                {
-                    logger.LogError($"Unable to analyze {filesFailed.Count} file(s): {string.Join(", ", filesFailed)}");
-                    return (int)(issueReported ? ExitCode.ErrorAndViolation : ExitCode.ErrorGeneric);
-                }
-                return issueReported ? (int)ExitCode.Violation : (int)ExitCode.Success;
-            }
-            catch (Exception exception)
-            {
-                logger.LogError(exception, "An exception occurred while analyzing the directory provided");
-                return (int)ExitCode.ErrorGeneric;
-            }
-            finally
-            {
-                this.summaryLoggerProvider.SummaryLogger.SummarizeLogs(verbose);
+                this.templateAnalyzer.FilterRules(config);
             }
+
+            return ExitCode.Success;
         }
 
-        private void FindJsonFilesInDirectoryRecursive(DirectoryInfo directoryPath, List<FileInfo> files)
+        private void FinishAnalysis()
         {
-            foreach (FileInfo file in directoryPath.GetFiles())
-            {
-                if (file.Extension.Equals(".json", StringComparison.OrdinalIgnoreCase))
+            this.reportWriter?.Dispose();
+        }
+
+        private IEnumerable<FileInfo> FindTemplateFilesInDirectory(DirectoryInfo directoryPath) =>
+            directoryPath.GetFiles(
+                "*.json",
```

Comment:
In the PR, you said it does not need to be a ".json" ending. Am I understanding this correctly that you are getting-the-files with a ".json" ending only?

Reply:
Analyze-template doesn't require a .json extension. This function is only called for analyze-directory.

Reply:
I see, that is why you specifically specified "Analyze-Template". I am sorry about that. 
Follow up, why only Analyze-Template and not Analyze-Directory too?
I know the size is an issue and can impact the runtime, but want to know more. Also on that note, should any Readme files be updated with this addition/change?

Reply:
Thanks for the README reminder - there was a separate change I wanted to include that I missed.  But for the .json change, we don't specify a requirement for that extension when scanning a template, so I don't think there's a reason for a change.  I might add something in the directory scan that explains it looks for .json files though.

.json is the standard ARM template extension, so if BPA is performing a directory search on its own, there's no reason to look at every single file to see if it happens to be a template; it should just stick to the standard.  On the other hand, if BPA is invoked to scan a *specific file* on the CLI (and there would only be a single file every time), my thought is there's no reason to *not* try to scan it.

Reply:
Thumbs-up to first paragraph.

2nd paragraph: It is fine now, sure, but like if the ARM template file is a .txt file instead of a .json file, someone using the tool could get the wrong impression when they use our tool that their .txt file is great. Then when they go to deploy it, the .txt file won't work and they could be confused.

Reply:
It depends on what you mean by "when they use our tool that their .txt file is great".  Analyze-template will still verify the file is actually a valid ARM template, so if the file is just plain text, it will still return an error.  It just won't require the file extension.

Reply:
I agree with Anais, if deploying a template using a .txt file returns an error I think we should too

Reply:
I just confirmed Azure Powershell will deploy a template file that has a ".txt" extension (it used a .txt parameters file too).  Can I leave this as-is and not require a .json extension?